R/make-array.R

```diff
@@ -330,3 +330,67 @@ flipArrays <- function(variables, suffix = ", flipped") {
     ## Return the list of derivations. Can then pass that to addVariables
     return(newvars)
 }
+
+#' Form a Multiple Response Variable from functions of existing variables
+#'
+#' Multiple Response Variables are a type of array variables (a set of "subvariables"
+#' bound together for display in the app) that take only 2 values ("selected" and "not").
+#' This function forms an MR variable by taking functions of existing variables and 
+#' deriving them at the same time as the new variable.
+#'
+#' @param name character, the name that the new Categorical Array variable should have.
+#' @param ... `SubvariableDefinition`s that will be the subvariables of the new variable
+#' @param .categories (Optional) Categories to define, defaults to the default MR 
+#' category (1 = "Selected", 0 = "Other")
+#' @param .subvariables A list of `SubvariableDefinition`s, like what would be 
+#' passed into `...`.
+#'
+#' @return A `VariableDefinition` that when added to a Dataset will create the multiple-response
+#' array and subvariables.
+#' @export
+formMR <- function(name, ..., .categories = NULL, .subvariables = NULL) {
+    if (is.null(.categories)) {
+        .categories <- Categories(
+            Category(id = 1L, missing = FALSE, name = "Selected", numeric_value = 1L, selected = TRUE),
+            Category(id = 0L, missing = FALSE, name = "Other", numeric_value = 0L),
+            Category(id = -1L, missing = TRUE, name = "No Data", numeric_value = NULL)
+        )
```

Comment:
great. leave this here in case we discover some reason to override it. dunno why, but good forethought.